aquasecurity/tracee
# [FEAT] Args syscall filter

## Initial Checklist

- [x] There is an issue describing the need for this PR (#2177).

## Description (git log)

commit 2182908a7ff0fd8c8153624e02f7c591a5249e08 (HEAD -> args_syscall_filter, origin/args_syscall_filter)
```
    filters: parse syscall argument filters by name
    
    This allows generic usage of syscall event names in relevant argument
    filters.
    
    Without this change, syscall ids may differ between architectures.
```

commit 7b2a68f3e06e434663db2c526f2c3be091fd7725
```
    filters: add generic Filter method to interface
    
    1. Add the generic Filter method, this takes any value to the filter.
    In practice, the external method must confirm the type, then pass to an
    internal type-safe method.
    
    2. Add generic type constraints to int and uint filters. This enables
    later down the line to avoid correct type casting for arguments.
```

## Type of change


## How Has This Been Tested?

1) Run `tracee-ebpf -t security_file_open.syscall=execve -o option:detect-syscall -t e=security_file_open`
Confirm that events are outputted only with "execve" syscalls and that most events are filtered:
`Stats: {EventCount:46 EventsFiltered:1322 NetEvCount:0 ErrorCount:0 LostEvCount:0 LostWrCount:0 LostNtCount:0}`

2) Run `tracee-ebpf -t task_rename.syscall=execnotve`
Confirm that tracee fails running.

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.


## Git Log Checklist:

My commits logs have:
